dbms/src/Interpreters/InterpreterDropQuery.cpp

```diff
@@ -98,11 +98,11 @@ BlockIO InterpreterDropQuery::execute()
 		}
 		else
 		{
-			/// Удаляем метаданные и саму таблицу из оперативки.
+			/// Delete table data
+			table.first->drop();
+			/// Delete table metdata and table itself from memory
 			database->removeTable(current_table_name);
 
-			/// Удаляем данные таблицы
-			table.first->drop();		/// TODO Не удалять метаданные, если таблицу не получилось удалить.
```

Comment:
Ok, You described previous behavior.
Now metadata will be removed if data was dropped successfully.
If we assume that metadata always removed successfully in `database->removeTable()` then current behavior doesn't have obvious drawbacks.